src/entity/Invite.ts

```diff
@@ -0,0 +1,32 @@
+import { Entity, PrimaryColumn, Column, ManyToOne, Index } from 'typeorm';
+import { User } from './User';
+import { FeatureType } from './Feature';
+
+export enum CampaignType {
+  Search = 'search',
+}
+
+@Entity()
+@Index(['userId', 'campaign'], { unique: true })
+export class Invite {
+  @PrimaryColumn({ type: 'text' })
+  token: string;
+
+  @Column({ type: 'text' })
+  campaign: CampaignType | FeatureType;
+
+  @Column({ length: 36 })
+  userId: string;
```

Comment:
You can define indexes here, it will generate neater once then if you run the creation.
Same for campaign field.

Reply:
Good point, however, this entity has a composite index on both `userId` and `campaign`, does defining indexes inline accomplish the same result? I thought it generates an index just for the respective column when defined here.